src/ansiblelint/file_utils.py

```diff
@@ -242,7 +242,7 @@ def __init__(
         self.abspath = self.path.expanduser().absolute()
 
         if self.kind == "yaml":
-            self._guess_kind()
+            self.data
```

Comment:
I'm only here by chance and don't know much about the internals of ansible-lint, but this statement is meaningless from my point of view.

Reply:
But it is not. The self.data function is "filling" self._data at the first call and is returning self._data the next calls.

Reply:
Is this a function and need `()` or does it use `__getattr__()`?

Reply:
The only reason why I did not merge this fix yet is because it does not update the tests. We need a reproducer for original bug as a test in order to ensure that next refactoring does not bring the bug back.

If you do not have time to write it, I will probably try to push the test myself.

Reply:
Oops. Sorry for that. I'll try to write it before the end of the week.